### PR TITLE
add simple concurrency and fix bug aboud compact and gc

### PR DESCRIPTION
**Pull request by eatbreads**
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Reference the link of issue using `fixes eloqdb/eloqstore#issue_id`
- [ ] Reference the link of RFC if exists
- [ ] Pass `ctest --test-dir build/tests/`

## Release Notes

* **Bug Fixes**
  * Enhanced memory eviction logic with additional validation to prevent data removal under unsafe conditions
  * Improved compact scheduling to properly handle edge cases with zero mappings

* **Tests**
  * Added concurrency soak test to validate system stability and reliability under concurrent load scenarios

### CODE REVIEW

File: test_utils.h [hunk 1]
```diff
@@ -152,6 +154,32 @@ class ConcurrencyTester
...
+    SimpleConcurrencyTest(eloqstore::EloqStore *store,
+                          std::chrono::milliseconds runtime,
+                          uint32_t partitions,
+                          uint32_t worker_threads = 0,
+                          uint32_t keys_per_partition = 256);
+    void Run();
+
+private:
+    void WorkerLoop(std::chrono::steady_clock::time_point deadline);
+    // void Drain(std::vector<std::unique_ptr<eloqstore::KvRequest>> &inflight);
```
Comment (thweetkomputer) [inline test_utils.h:169]: remove
